core/src/main/scala/org/bitcoins/core/wallet/builder/RawTxFinalizer.scala

```diff
@@ -182,6 +137,106 @@ object NonInteractiveWithChangeFinalizer {
       Success(())
     }
   }
+}
+
+/** A finalizer which performs fee estimation and adds a
+  * change output resulting in the expected fee.
+  */
+case class ChangeFinalizer(
+    inputInfos: Vector[InputInfo],
+    feeRate: FeeUnit,
+    changeSPK: ScriptPubKey)
+    extends RawTxFinalizer {
+  override def buildTx(txBuilderResult: RawTxBuilderResult)(
+      implicit ec: ExecutionContext): Future[Transaction] = {
```

Comment:
Should add dummy witness data for `txBuilderResult` that will result in witness txs.